src/app/qgsdatasourcemanagerdialog.h

```diff
@@ -0,0 +1,74 @@
+/***************************************************************************
+    qgsdatasourcemanagerdialog.h - datasource manager dialog
+
+    ---------------------
+    begin                : May 19, 2017
+    copyright            : (C) 2017 by Alessandro Pasotti
+    email                : apasotti at itopen dot it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSDATASOURCEMANAGERDIALOG_H
+#define QGSDATASOURCEMANAGERDIALOG_H
+
+#include <QList>
+#include <QDialog>
+#include "qgsoptionsdialogbase.h"
+#include "qgsguiutils.h"
+
+class QgsBrowserDockWidget;
+class QgsRasterLayer;
+class QgsMapCanvas;
+
+namespace Ui
+{
+  class QgsDataSourceManagerDialog;
+}
+
+class QgsDataSourceManagerDialog : public QgsOptionsDialogBase
```

Comment:
@nyalldawson @wonder-sk I'm looking at moving this class to GUI, but I've found more dependencies, like `qgsnewogrconnection.h` and all its companions under `app/ogr`, do you think that worths the effort moving all ogr deps to GUI too? I start to doubt about that.